### ISSUE TITLE
docs(inference): correct three public API signatures and document the extension seams

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#240]
Part of the docs accuracy epic. **Content only** — do not change any Python code in this issue.

Target file: `radiologist-inference/README.md`

`docs/pkg/inference.md` is a pure `{% include-markdown %}` shim, so editing the README is the whole job.

This README is in good shape overall — the unified-CLI migration is already done (the `### CLI` section correctly says the package ships no console script, and `pyproject.toml` has no `[project.scripts]`), and `### Library` already covers programmatic use well. It needs three factual corrections and one new section.

---

### 1. `explain` signature is incomplete (`## Public API reference`, ~line 212)

Documented as `explain(image) -> Explanation`. The real signature at `src/radiologist/inference/explainer.py:41` is:

```python
def explain(
    self,
    image: Union[str, "np.ndarray", "PILImage.Image"],
    deployment_prior: Optional[Dict[str, float]] = None,
) -> Explanation:
```

It takes `deployment_prior` just like `predict` does.

### 2. The `BasePredictor` method table omits two public members (~lines 197–200)

Missing:

- `from_selector(selector, local_dir, registry=None, mean=None, std=None, input_shape=None) -> BasePredictor` (`base_predictor.py:170`)
- the `provenance -> Optional[ArtifactRef]` property (`base_predictor.py:72`)

Both are public and both are what the CLI actually uses to record which artifact a prediction came from.

### 3. `score_cam`'s `logits` argument is described as if it were read (~line 226)

The README says it takes `logits (num_classes,)`. `cam.py`'s own docstring (lines 38–45) states `logits` is **unused**, retained only for interface symmetry.

### 4. There is no "extending" content

Verified: this package has **zero** yaml files, no `conf/`/`config/`/`configs/` directory, no `ConfigStore`/`cs.store`, and no `hydra` import anywhere in `src`. Unlike core and etl, there is genuinely nothing to point a `hydra.searchpath` at — and the docs should say so plainly rather than leave a reader hunting for a config tree that does not exist.

---

### AC1 — Three signature/description corrections

- Change the `explain` row to `explain(image, deployment_prior=None) -> Explanation`.
- Add two rows to the `BasePredictor` table:
  - `` `from_selector(selector, local_dir, registry=None, mean=None, std=None, input_shape=None)` `` — build a predictor from a `RegistrySelector`; this is the path the CLI takes.
  - `` `provenance` `` (property) — `Optional[ArtifactRef]` recording which registry artifact this predictor was loaded from; `None` for a predictor built with `from_path`.
- Amend the `score_cam` description: "`logits` is accepted for interface symmetry and is not read."

### AC2 — New `## Extending this package`

Insert **after** the `### Result dataclasses` subsection of `## Public API reference`, **before** `## Development setup`.

`````markdown
## Extending this package

**There are no user-swappable Hydra config groups here.** Unlike
`radiologist-core` and `radiologist-etl`, this package ships no config tree and
does not depend on Hydra at all — the `radiologist infer` commands are plain
Typer flags, not `key=value` config overrides. There is nothing to point a
`hydra.searchpath` at.

Two real extension seams exist instead.

### Swapping the model registry backend

Every registry-backed loader takes an optional `registry` argument and defaults
to `WandbRegistry()`. Pass anything satisfying the
[`ModelRegistry`](../radiologist-registry/README.md#extending-the-registry-backend)
protocol — it is a structural `typing.Protocol`, so no inheritance or
registration is needed:

```python
from radiologist.inference import Classifier

predictor = Classifier.from_registry(
    artifact_path="entity/project/model-abc123:best",
    local_dir="/tmp/models",
    registry=MyRegistry(),   # any object matching radiologist.registry.ModelRegistry
)
```

This is the seam to use for an on-prem artifact store, a test double, or a
caching layer in front of W&B.

### Adding a verb by subclassing

`BasePredictor` owns loading, provenance, and preprocessing; each subclass adds
one verb (`Classifier.predict`, `Explainer.explain`,
`MCDropoutPredictor.predict_with_uncertainty`). Construction goes through
`cls.__new__(cls)` inside the three classmethods, so a subclass inherits
`from_path` / `from_registry` / `from_selector` for free:

```python
from radiologist.inference import Classifier


class TopKClassifier(Classifier):
    def predict_top_k(self, image, k: int = 3):
        prediction = self.predict(image)
        ranked = sorted(prediction.probabilities.items(), key=lambda kv: -kv[1])
        return ranked[:k]


predictor = TopKClassifier.from_path("/tmp/models/model.onnx")
```

The HTTP layer is not currently extensible in the same way: `create_app`
dispatches routes by `isinstance` over the three shipped predictor classes, so a
subclass gets its parent's routes, not a new one. The internal verb table in
`radiologist.inference.verbs` is module-private with no registration function —
it is not a supported extension point today.
`````

---

## Notes for the implementer

- No `python -m` staleness to fix here — I checked. Do not "helpfully" rewrite the `### CLI` section; it is already correct, including the `../docs/reference/cli-inference.md` link, which resolves correctly from the README's location.
- All Python examples are 3.10-compatible. Note that **no file in this package uses `X | Y` union syntax**, even the four that carry `from __future__ import annotations` (`base_predictor.py`, `classifier.py`, `explainer.py`, `mc_dropout.py`) — keep documenting with `Optional[...]` / `Union[...]` to match.

## Definition of done

- [ ] Both ACs applied to `radiologist-inference/README.md`
- [ ] No source file changed
- [ ] `mkdocs build` succeeds (`make docs-build`)
